### ISSUE TITLE
[BUG] ObjectReplicationSourceProperties is null

### ISSUE THREAD

**Issue by asgoe** [Azure/azure-sdk-for-net#26140]
### Library name and version

Azure.Storage.Blobs v12.10.0

### Describe the bug

We are testing the Object Replication feature for blobs. Based on this [link ](https://docs.microsoft.com/en-us/azure/storage/blobs/object-replication-configure?tabs=powershell#check-the-replication-status-of-a-blob) we should be able to check the replication status of the blobs through the ObjectReplicationSourceProperties. However, the property is coming as null for all blobs. As a result, blob.ObjectReplicationSourceProperties[0].Rules[0].ReplicationStatus does not work and throws null reference exception. We are testing this on an account where the replication has already taken place and the replicated blobs can be seen in the destination account. However, the ObjectReplicationSourceProperties is coming as null for blobs in both source and destination accounts. In the Azure portal, the replication properties are visible.

### Expected behavior

ObjectReplicationSourceProperties[0].Rules[0].ReplicationStatus should return complete or failure for blobs in storage accounts where replication policy has been applied.

### Actual behavior

ObjectReplicationSourceProperties is coming as null.

### Reproduction Steps

1) Followed the steps in this page for the portal to create the replication policies: https://docs.microsoft.com/en-us/azure/storage/blobs/object-replication-configure?tabs=portal

2) Used the code below to list and iterate through the blobs. Replace the variables in <> for the account, container, access key
```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Azure;
using Azure.Storage;
using Azure.Storage.Blobs;
using Azure.Storage.Blobs.Models;

namespace TestBlobReplication
{
    class Program
    {
        //Azure storage account information
        private static readonly string storageAccountName = <**STORAGE ACCOUNT NAME**>;
        private static readonly string storageContainerName = <**CONTAINER NAME**>;
        private static readonly string AzureStoragePrimaryKey = <**PRIMARY ACCESS KEY**>;
        private static readonly string AzureStorageConnectionString = $"DefaultEndpointsProtocol=https;AccountName={storageAccountName};AccountKey={AzureStoragePrimaryKey};EndpointSuffix=core.windows.net";

        private static BlobContainerClient blobContainerClient;

        public async Task<BlobContainerClient> GetBlobContainerReferenceAsync(string containerName)
        {
            var blobContainerClient = new BlobContainerClient(AzureStorageConnectionString, containerName);
            bool exists = await blobContainerClient.ExistsAsync();

            if (exists)
            {
                Console.WriteLine($"{DateTime.Now} \n2. Found Container \n{containerName}\n");
            }
            else
            {
                await blobContainerClient.CreateAsync();
                Console.WriteLine($"{DateTime.Now} Blob container: {containerName} inside storage account: {blobContainerClient.AccountName} does not exist, container created");
                //throw new InvalidOperationException("Storage Account/Container not found");
            }

            return blobContainerClient;
        }

        public async Task<IList<BlobItem>> ListBlobMetadataAsync(
            string prefix,
            string continuationToken,
            int maxResultCount = 5000)
        {
            // TODO: make this.container auto initialize   
            IAsyncEnumerable<Page<BlobItem>> blobPages = blobContainerClient.GetBlobsAsync(
                BlobTraits.Metadata,
                BlobStates.None,
                prefix)
                .AsPages(continuationToken, maxResultCount);

            IAsyncEnumerator<Page<BlobItem>> enumerator = blobPages.GetAsyncEnumerator();
            IList<BlobItem> resultSegment = new List<BlobItem>();
            string newContinuationToken = null;

            try
            {
                while (await enumerator.MoveNextAsync())
                {
                    Page<BlobItem> blobPage = enumerator.Current;
                    foreach (BlobItem blobItem in blobPage.Values)
                    {
                        resultSegment.Add(blobItem);
                    }
                    newContinuationToken = blobPage.ContinuationToken;
                }
            }
            finally
            {
                await enumerator.DisposeAsync();
            }

            return resultSegment;
        }

        static async Task Main(string[] args)
        {
            Console.WriteLine($"{DateTime.Now} Beginning operations...");
            Program p = new Program();

            blobContainerClient = await p.GetBlobContainerReferenceAsync(storageContainerName);
            //BlobClient blobClient = blobContainerClient.GetBlobClient(blobId);
            IList<BlobItem> blobs = await p.ListBlobMetadataAsync(null, null);

            foreach (var blob in blobs)
            {
                var metadata = blob.Metadata;
                var createdTime = blob.Properties.CreatedOn;
                var replicated = blob.ObjectReplicationSourceProperties[0].Rules[0].ReplicationStatus;
            }
        }
    }
}
```

### Environment

.NET CORE 3.1
Visual Studio 2019

**Comment by SaurabhSharma-MSFT** [comment]
@asgoe I see that you are using `GetBlobsAsync` in your code.   You may need to use `DownloadAsync()` or `DownloadContentAsync()` to get the ObjectReplicationSourceProperties.  (See below)
![image](https://user-images.githubusercontent.com/38112130/147993884-5d912073-df42-4351-b1bd-5a3720cb4cb1.png)

Thanks
Saurabh

**Comment by asgoe** [comment]
Thanks @SaurabhSharma-MSFT, we are trying to find a way to validate the replication for all the blobs in a storage account for data move. Are recommendation on what would be the best way to do so? As of now it seems that we will have to list all the blobs, filter out based on last modified time and then download and check the replication status. This can be a slow process if our account has a large number of blobs, especially considering that we will be taking a customer downtime when we perform this validation.

**Comment by asgoe** [comment]
@SaurabhSharma-MSFT I am still getting the ObjectReplicationSourceProperties as null after calling the download operation on the blob
See below, lines 93-95 is the code that I added to download the blobs that are received as part of the list operation:
![image](https://user-images.githubusercontent.com/58997343/148110644-5f2becc3-7125-4c7d-b5e7-59d793da723a.png)


This is the screenshot of the blob's properties from the portal:
![image](https://user-images.githubusercontent.com/58997343/148110361-743b4fb3-7f2b-4d5c-b100-d3b5f21b1b61.png)

**Comment by asgoe** [comment]
@SaurabhSharma-MSFT any feedback on ObjectReplicationSourceProperties still coming as null?

**Comment by SaurabhSharma-MSFT** [comment]
@asgoe 
![image](https://user-images.githubusercontent.com/38112130/148440145-6fc90cad-18bc-453d-a2eb-a0412aa0594f.png)
I agree with you that you don't need to download the blob every time to check for this property and this should be returned as part of `GetBlobsAsync`.  I am redirecting this issue to service team to look into this and guide with your Null issue with `DownloadAsync` method. Thank you.

**Comment by asgoe** [comment]
Can I get some feedback on this? This is blocking our design.

**Comment by asgoe** [comment]
@jsquire, @xgithubtriage, its been more than 2 weeks without any action on this. Can this please be assigned to the correct team? This is a blocker for us.

**Comment by jsquire** [comment]
Hi @asgoe.   I'm sorry to hear that there hasn't been further engagement.  Unfortunately, this has already been triaged to the correct team and I have no insight into its current state.    @amishra:  Is there any insight that you're able to provide?

**Comment by amishra-dev** [comment]
Tagged the wrong person @jsquire Jesse :)

@asgoe, lets sort it out over email and we can put the summary here. I do apologize for the delay

**Comment by jsquire** [comment]
> Tagged the wrong person @jsquire Jesse :)

Apologies; that's my mistake.

**Comment by amnguye** [comment]
> @SaurabhSharma-MSFT I am still getting the ObjectReplicationSourceProperties as null after calling the download operation on the blob See below, lines 93-95 is the code that I added to download the blobs that are received as part of the list operation: ![image](https://user-images.githubusercontent.com/58997343/148110644-5f2becc3-7125-4c7d-b5e7-59d793da723a.png)
> 
> This is the screenshot of the blob's properties from the portal: ![image](https://user-images.githubusercontent.com/58997343/148110361-743b4fb3-7f2b-4d5c-b100-d3b5f21b1b61.png)

Also regarding this issue, it's expected behavior that the service won't return back a `ObjectReplicationSourceProperties` in `BlobDownloadInfo` when you go to download the destination blob. If you go to download the source blob, the `BlobDownloadInfo.ObjectReplicationSourceProperties` will be populated, and the `BlobDownloadInfo.ObjectReplicationDestinationPolicyId` will be null.

**Comment by amishra-dev** [comment]
Fixed in 12.11 (https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/CHANGELOG.md#12110-beta1-2021-11-03)
